engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/Builtins.java

```diff
@@ -59,27 +63,21 @@ public static class Debug {
     }
   }
 
-  private final AtomConstructor any;
+  private HashMap<String,Map<String, Class<BuiltinRootNode>>> builtinNodes;
+  // TODO Consider dropping the map and just assigning to a single variable since builtin types
+  //      should be unique
+  private HashMap<String, AtomConstructor> builtinTypes;
```

Comment:
This is going to cause major slowdown. Need to either revert to the old approach or come up with something generic that doesn't bring **huge** performance penalty.